### ISSUE TITLE
Crash when Entering decimal point in a blank field

### ISSUE THREAD

**Issue by adcaine** [BlacKCaT27/CurrencyEditText#37]
Reproduction Steps:

In the sample app,

1.  Click in Reset button.
2.  Type a decimal point in the first field

Expected

3.  The string "$0." appears in the field

Actual

The app crashes.

09-23 20:08:43.367 14567-14567/com.blackcat.currencyedittexttester E/AndroidRuntime: FATAL EXCEPTION: main
                                                                                     Process: com.blackcat.currencyedittexttester, PID: 14567
                                                                                     java.lang.IndexOutOfBoundsException: setSpan (1 ... 1) ends beyond length 0
                                                                                         at android.text.SpannableStringBuilder.checkRange(SpannableStringBuilder.java:1265)
                                                                                         at android.text.SpannableStringBuilder.setSpan(SpannableStringBuilder.java:684)
                                                                                         at android.text.SpannableStringBuilder.setSpan(SpannableStringBuilder.java:677)
                                                                                         at android.text.Selection.setSelection(Selection.java:76)
                                                                                         at android.widget.TextView.semSetSelection(TextView.java:11458)
                                                                                         at android.widget.TextView.semSetSelection(TextView.java:11472)
                                                                                         at android.widget.EditText.setSelection(EditText.java:124)
                                                                                         at com.blackcat.currencyedittext.CurrencyTextWatcher.afterTextChanged(CurrencyTextWatcher.java:68)
                                                                                         at android.widget.TextView.sendAfterTextChanged(TextView.java:9443)

**Comment by BlacKCaT27** [comment]
Thanks for the report! I have a fix in the works which should be pushed up
to the 'next' branch shortly.

I can't match the expected behavior of forcing a partially formatted number
like "$0." to appear, as that's not how the formatter works and would
require some hackiness peppered around the formatting code, but since this
error also occurs with '," and "-" characters, I've simply stopped the code
causing the crash (the cursor placement code) from executing if the string
length is zero. Perhaps not ideal, but it's what we've got.
